controllers/tournament.controller.js

```diff
@@ -0,0 +1,256 @@
+                message: "You are not a member of this club",
+            });
+        }
+
+        const clubTournaments = await tournaments.find({
+            clubId: new ObjectId(req.params.clubId),
+        }).sort({ created: -1 }).toArray();
+
+        res.status(200).json(clubTournaments);
+    } catch (error) {
+        console.error("Error during getting club tournaments:", error);
+        return res.status(500).json({ message: "Internal server error" });
+    }
+}
+
+const UPDATE_SCHEMA = z.object({
+    name: z.string().trim().min(3).max(256).optional(),
+    startTime: z.string().optional(), //i dont know how to convert to date (they use datetime() but its deprecated)
+    status: z.enum(["draft", "active", "completed"]).optional(),
+    settings: z.object({
+        rankingConfig: z.object({
+            winPoints: z.number().min(0).optional(),
+            drawPoints: z.number().min(0).optional(),
+            lossPoints: z.number().min(0).optional(),
+            addScorePoints: z.boolean().optional(),
+        }).optional(),
+    }).optional(),
+}).strict();
+
+/** @type {import("express").RequestHandler<{ tournamentId: string }>} */
+export async function updateTournament(req, res) {
+    try {
+        const parsed = UPDATE_SCHEMA.parse(req.body);
+
+        const tournament = await tournaments.findOne({
+            _id: new ObjectId(req.params.tournamentId),
+        });
+
+        if (tournament == null) {
+            return res.status(404).json({
+                message: "Tournament not found",
+            });
+        }
+
+        // Verify user has permission (owner or admin)
+        const membership = await clubMembers.findOne({
+            clubId: tournament.clubId,
+            userId: new ObjectId(req.user.id),
+            role: { $in: ["owner", "admin"] },
+        });
+
+        if (membership == null) {
+            return res.status(403).json({
+                message: "You don't have permission to update this tournament",
+            });
+        }
+
+        const updateDoc = {};
+        if (parsed.name) updateDoc.name = parsed.name;
+        if (parsed.startTime) updateDoc.startTime = new Date(parsed.startTime);
+        if (parsed.status) updateDoc.status = parsed.status;
+        if (parsed.settings?.rankingConfig) {
+            updateDoc["settings.rankingConfig"] = {
+                ...tournament.settings?.rankingConfig,
+                ...parsed.settings.rankingConfig,
+            };
+        }
+
+        const { modifiedCount } = await tournaments.updateOne(
+            { _id: new ObjectId(req.params.tournamentId) },
+            { $set: updateDoc }
+        );
+
+        if (modifiedCount === 0) {
+            return res.status(400).json({
+                message: "No changes were made",
+            });
+        }
+
+        const updatedTournament = await tournaments.findOne({
+            _id: new ObjectId(req.params.tournamentId),
+        });
```

Comment:
Could be reduced to only one call by using `tournaments.findOneAndUpdate` instead of `tournaments.updateOne`:

```ts
tournaments.findOneAndUpdate(query, updates, { returnDocument: "after" })
```

returns the modified document.